lib/yardstick/decorator.rb

```diff
@@ -0,0 +1,60 @@
+module Yardstick
+  # A string decorator for applying unix console codes
+  class Decorator
+    FORMAT = "\e[%{mode};%{color}%{string}\e[0m".freeze
+
+    COLOR_CODES = {
+      red:    31,
+      yellow: 33
+    }.freeze
+
+    MODE_CODES = {
+      bold:      1,
+      underline: 4
+    }.freeze
+
+    private_constant(:FORMAT, :COLOR_CODES, :MODE_CODES)
+
+    # Initializes new string decorator instance
+    #
+    # @param color [Symbol] name of color to use for formatting
+    # @param mode [Symbol] mode for formatting text
+    #
+    # @return [undefined]
+    #
+    # @api private
+    def initialize(color, mode)
+      @color = COLOR_CODES.fetch(color)
+      @mode  = MODE_CODES.fetch(mode)
+    end
```

Comment:
If you'd use concord you can reduce this a bit.

```
include Concord(:color, :mode)

def self.new(color, mode)
  super(COLOR_CODES.fetch(color), MODE_CODES.fetch(mode))
end
```

It will generate the attr readers below, and you do have less chance to typo the ivars.